ipatests/test_integration/test_adtrust_install.py

```diff
@@ -0,0 +1,190 @@
+#
+# Copyright (C) 2020  FreeIPA Contributors see COPYING for license
+#
+
+"""This module provides tests for ipa-adtrust-install utility"""
+
+import re
+import textwrap
+
+from ipaplatform.paths import paths
+from ipapython.dn import DN
+from ipatests.pytest_ipa.integration import tasks
+from ipatests.test_integration.base import IntegrationTest
+
+
+class TestIpaAdTrustInstall(IntegrationTest):
+    topology = 'line'
+    num_replicas = 1
+
+    def unconfigure_replica_as_agent(self, host):
+        """ Remove a replica from the list of agents.
+
+        cn=adtrust agents,cn=sysaccounts,cn=etc,$BASEDN contains a list
+        of members representing the agents. Remove the replica principal
+        from this list.
+        """
+        remove_agent_ldif = textwrap.dedent("""
+             dn: cn=adtrust agents,cn=sysaccounts,cn=etc,{base_dn}
+             changetype: modify
+             delete: member
+             member: fqdn={hostname},cn=computers,cn=accounts,{base_dn}
+             """.format(base_dn=host.domain.basedn, hostname=host.hostname))
+        # ok_returncode =16 if the attribute is not present
+        tasks.ldapmodify_dm(self.master, remove_agent_ldif,
+                            ok_returncode=[0, 16])
+
+    def test_samba_config_file(self):
+        """Check that ipa-adtrust-install generates sane smb.conf
+        This is regression test for issue
+        https://pagure.io/freeipa/issue/6951
+        """
+        self.master.run_command(
+            ['ipa-adtrust-install', '-a', self.master.config.admin_password,
+             '--add-sids', '-U'])
+        res = self.master.run_command(['testparm', '-s'])
+        assert 'ERROR' not in (res.stdout_text + res.stderr_text)
+
+    def test_add_agent_not_allowed(self):
+        """Check that add-agents can be run only by Admins
+        """
+        user = "nonadmin"
+        passwd = "Secret123"
+        tasks.create_active_user(self.master, user, passwd,
+                                 first=user, last=user)
+        tasks.kinit_as_user(self.master, user, passwd)
+        host = self.replicas[0].hostname
+        data_fmt = '{{"method":"trust_enable_agent","params":[["{}"],{{}}]}}'
+        cmd_args = [
+            paths.BIN_CURL,
+            '-H', 'referer:https://{}/ipa'.format(host),
+            '-H', 'Content-Type:application/json',
+            '-H', 'Accept:applicaton/json',
+            '--negotiate', '-u', ':',
+            '--cacert', paths.IPA_CA_CRT,
+            '-d', data_fmt.format(host),
+            '-X', 'POST', 'https://{}/ipa/json'.format(host)]
+        res = self.master.run_command(cmd_args)
+        tasks.kinit_admin(self.master)
+        expected_re = 'Insufficient access: not allowed to remotely add agent'
+        assert re.search(expected_re, res.stdout_text)
```

Comment:
No need to use re.search here as the string does not contain any regexp operators

Reply:
OK, updated.